src/odemis/gui/cont/tabs/correlation_tab.py

```
@@ -167,10 +171,14 @@ def _on_add_tileset(self) -> None:
         self.select_acq_file(extend=True, tileset=True)
 
     def load_tileset(self, filenames: List[str], extend: bool = False) -> None:
+        # Convert all files to pyramidal if they're TIFFs
+        filenames = ensure_pyramidal_tiff_for_tileset_gui(filenames, self.panel, self.main_data)
         data = open_files_and_stitch(filenames) # TODO: allow user defined registration / weave methods
         self.load_streams(data)
 
     def load_data(self, filename: str, fmt: str = None, extend: bool = False) -> None:
```

Ruff (0.15.6)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)